application-xpoll-api/src/main/java/com/xwiki/xpoll/internal/DefaultXPollManager.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.xpoll.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.model.reference.LocalDocumentReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.xpoll.XPollManager;
+
+public class DefaultXPollManager implements XPollManager
+{
+    private static final String SPACENAME = "XPoll";
+    private static final LocalDocumentReference XPOLL_CLASSREFERENCE =
+        new LocalDocumentReference(SPACENAME, "XPollClass");
+
+    private static final LocalDocumentReference XPOLL_VOTES_CLASSREFERENCE =
+        new LocalDocumentReference(SPACENAME, "XPollVoteClass");
+
+    private static final String STATUS = "status";
+
+    private static final String PROPOSALS = "proposals";
+
+    private static final String VOTES = "votes";
+
+    private static final String WINNER = "winner";
+
+    private static final String USER = "user";
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public void execute(XWikiDocument page, Map<String, String> votedProposals, XWikiContext context)
+    {
+        try {
+            BaseObject xpollObj = page.getXObject(XPOLL_CLASSREFERENCE);
+
+            List<String> proposals = xpollObj.getListValue(PROPOSALS);
+
+            setUserVotes(votedProposals, context, page, proposals);
+            updateWinner(context, page, xpollObj, proposals);
+        } catch (XWikiException e) {
+            logger.error(e.getFullMessage());
+        }
+    }
+
+    private void setUserVotes(Map<String, String> votedProposals, XWikiContext context, XWikiDocument doc,
+        List<String> proposals) throws XWikiException
+    {
+        String currentUserName = context.getUserReference().getLocalDocumentReference().toString();
+        BaseObject xpollVoteOfCUrrentUser = doc.getXObject(XPOLL_VOTES_CLASSREFERENCE, USER,
+            currentUserName, false);
+        Map<String, String> sortedVotes = new TreeMap<>();
+
+        List<String> votes = new ArrayList<>();
+        for (String proposal : proposals) {
+            String proposalHash = String.valueOf(proposal.hashCode());
+            String value = votedProposals.get(proposalHash);
```

Comment:
Thats the way it is done originally. Each proposal checkbox input has its name as a hash. 
`<input type="checkbox" #if($isCurrentUser)id="$proposal.hashCode()"#end name="$proposal.hashCode()" $checked $disabled />`
It didn't want to change it because I assumed the original creator had a reason for it, so I worked around it.